quickcache/quickcache_helper.py

```diff
@@ -0,0 +1,150 @@
+import hashlib
+import inspect
+from inspect import isfunction
+
+from .logger import logger
+
+
+class QuickCacheHelper(object):
+    def __init__(self, fn, vary_on, cache, skip_arg=None, assert_function=None):
+
+        self.fn = fn
+        self.cache = cache
+        self.prefix = '{}.{}'.format(
+            fn.__name__[:40] + (fn.__name__[40:] and '..'),
+            self._hash(inspect.getsource(fn), 8)
+        )
+
+        arg_names = inspect.getargspec(self.fn).args
+        if not isfunction(vary_on):
+            vary_on = [part.split('.') for part in vary_on]
+            vary_on = [(part[0], tuple(part[1:])) for part in vary_on]
+            for arg, attrs in vary_on:
+                if arg not in arg_names:
+                    raise ValueError(
+                        'We cannot vary on "{}" because the function {} has '
+                        'no such argument'.format(arg, self.fn.__name__)
+                    )
+
+        self.encoding_assert = assert_function
+
+        self.vary_on = vary_on
+
+        if skip_arg is None or isinstance(skip_arg, basestring) or isfunction(skip_arg):
+            self.skip_arg = skip_arg
+        else:
+            raise ValueError("skip_arg must be None, a string, or a function")
+
+        arg_spec = inspect.getargspec(self.fn)
+        if isinstance(skip_arg, basestring) and self.skip_arg not in arg_spec.args:
+            raise ValueError(
+                'We cannot use "{}" as the "skip" parameter because the function {} has '
+                'no such argument'.format(self.skip_arg, self.fn.__name__)
+            )
```

Comment:
This stuff was also adapted from `SkippablQuickCache` (RIP)'s `__init__` method.